lib/Repository.js

```diff
@@ -172,6 +172,19 @@ class Repository extends Requestable {
       return this._request('GET', `/repos/${this.__fullname}/commits`, options, cb);
    }
 
+    /**
+     * Gets a single commit information for a repository
+     * @see https://developer.github.com/v3/repos/commits/#get-a-single-commit
+     * @param {string} ref - the reference for the commit, if not provided this method behaves as listCommits
+     * @param {string} ref - the reference for the commit
+     * @param {Requestable.callback} cb - will receive the commit information
+     * @return {Promise} - the promise for the http request
+     */
+    getSingleCommit(ref, cb) {
+       ref = ref || '';
```

Comment:
It does return an object and it does fail. The replacement of ref with '' is just to make sure that if someone calls a branch 'undefined' or 'null' the API might return results when ref wasn't provided. See unit tests showing that it does fail.